Add tests for re-compile when doing loop osr in BBQ B3 mode.

justinmichaud opened the pull request:
#### 11024a1dd2a94ed4f31c2c1e4a34e5363dd7beb5
<pre>
Add tests for re-compile when doing loop osr in BBQ B3 mode.
<a href="https://bugs.webkit.org/show_bug.cgi?id=242294.">https://bugs.webkit.org/show_bug.cgi?id=242294.</a>

Air and B3 have two different approaches to loop osr. In BBQ Air, every
loop is included as a potential entry point, so we only need to compile
once. For OMG B3, we wish to compile with only one entry point at a time,
so that we do not pessimize loops.

Previously, we would always assume that BBQ generated all loop entry points,
even when we forced the use of BBQ B3. This happens on iOS for large modules
due to a workaround introduced to prevent executable memory exhaustion, and
also when we manually run BBQ without air.

This patch adds a test case and also adds extra test suite options to
test this unusual mode regularly.

Reviewed by Yusuke Suzuki.

* JSTests/wasm/wast-tests/osr-loop-entry-force-b3.wasm: Added.
* JSTests/wasm/wast-tests/osr-loop-entry-force-b3.wast: Added.
* Tools/Scripts/run-jsc-stress-tests:

Canonical link: <a href="https://commits.webkit.org/252171@main">https://commits.webkit.org/252171@main</a>
</pre>

Constellation reviewed (approved):
r=me. Let's land these tests after landing #2105

Commit:
Add tests for re-compile when doing loop osr in BBQ B3 mode.
https://bugs.webkit.org/show_bug.cgi?id=242294.

Air and B3 have two different approaches to loop osr. In BBQ Air, every
loop is included as a potential entry point, so we only need to compile
once. For OMG B3, we wish to compile with only one entry point at a time,
so that we do not pessimize loops.

Previously, we would always assume that BBQ generated all loop entry points,
even when we forced the use of BBQ B3. This happens on iOS for large modules
due to a workaround introduced to prevent executable memory exhaustion, and
also when we manually run BBQ without air.

This patch adds a test case and also adds extra test suite options to
test this unusual mode regularly.

Reviewed by Yusuke Suzuki.

* JSTests/wasm/wast-tests/osr-loop-entry-force-b3.wasm: Added.
* JSTests/wasm/wast-tests/osr-loop-entry-force-b3.wast: Added.
* Tools/Scripts/run-jsc-stress-tests:

Canonical link: https://commits.webkit.org/252171@main
